plugins/linux-swap/fu-plugin-linux-swap.c

```diff
@@ -36,41 +39,72 @@ fu_plugin_linux_swap_changed_cb (GFileMonitor *monitor,
 				 GFileMonitorEvent event_type,
 				 gpointer user_data)
 {
-	g_debug ("swap changed");
+	FuPlugin *plugin = FU_PLUGIN (user_data);
+	fu_plugin_security_changed (plugin);
 }
 
 gboolean
 fu_plugin_startup (FuPlugin *plugin, GError **error)
 {
 	FuPluginData *data = fu_plugin_get_data (plugin);
-	gsize bufsz = 0;
-	g_autofree gchar *buf = NULL;
 	g_autofree gchar *fn = NULL;
 	g_autofree gchar *procfs = NULL;
-	g_autoptr(FuLinuxSwap) swap = NULL;
-	g_autoptr(GFile) file = NULL;
 
 	procfs = fu_common_get_path (FU_PATH_KIND_PROCFS);
 	fn = g_build_filename (procfs, "swaps", NULL);
-	file = g_file_new_for_path (fn);
-	data->monitor = g_file_monitor (file, G_FILE_MONITOR_NONE, NULL, error);
+	data->file = g_file_new_for_path (fn);
+	data->monitor = g_file_monitor (data->file, G_FILE_MONITOR_NONE, NULL, error);
 	if (data->monitor == NULL)
 		return FALSE;
 	g_signal_connect (data->monitor, "changed",
 			  G_CALLBACK (fu_plugin_linux_swap_changed_cb), plugin);
+	return TRUE;
+}
 
-	/* load list of linux_swaps */
-	if (!g_file_get_contents (fn, &buf, &bufsz, error)) {
-		g_prefix_error (error, "could not open %s: ", fn);
-		return FALSE;
+void
+fu_plugin_add_security_attrs (FuPlugin *plugin, FuSecurityAttrs *attrs)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	gsize bufsz = 0;
+	g_autofree gchar *buf = NULL;
+	g_autoptr(FuLinuxSwap) swap = NULL;
+	g_autoptr(FwupdSecurityAttr) attr = NULL;
+	g_autoptr(GError) error_local = NULL;
+
+	/* create attr */
+	attr = fwupd_security_attr_new ("org.kernel.Swap");
```

Comment:
If this came in from the changed callback, (IE If the swap keeps changing) then don't we need to also evacuate the existing attribute

Reply:
No, the FuSecurityAttrs is created for each request.